docs/ref/models/instances.txt

```diff
@@ -589,6 +589,24 @@ row. In these cases you can pass the ``force_insert=True`` or
 Passing both parameters is an error: you cannot both insert *and* update at the
 same time!
 
+.. versionchanged:: 5.0
+
+    In case of an inherited model the ``force_insert=True`` only forces the
+    insertion in the leaf model. ``force_insert=False`` is taken as default for the
+    parent models.
+
+    If it is required to force insert the parent models also, then consider passing
+    a tuple of Model classes that requires to perform force insertion. For
+    example::
+
+        Child(id=3, name="Child-3", value=10).save(force_insert=(GrandParent, Parent))
```

Comment:
I think `(GrandParent, Parent)` is redundant with `GrandParent` as `issubclass(Parent, GrandParent)`?

Reply:
Yes, that's what I about to tell you as in Django one inherited model class cannot have more than one non-abstract parent class what is the point of passing a tuple, we can pass a single models.Model subclass.

Reply:
Tuples, are only necessary when dealing with diamond inheritance.

Say that you have

```python
class A(models.Model):
   pass

class B(A):
   pass

class C(A):
   pass

class D(B, C):
   pass
```

It might be interesting to `force_insert` `B` and `C` while not `A`! Only in this case does passing a tuple is necessary as that's only doable by doing `d.save(force_insert=(B, C))`.

Reply:
Tried this but a_ptr willbe clashing in D, so it won't be a valid  inheritance I think.

Reply:
`
class GrandParent(models.Model):
    name = models.CharField(max_length=50, null=True)
    value = models.IntegerField(null=True)


class Parent(GrandParent):
    pass


class DiamondParent(GrandParent):
    pin_code = models.CharField(max_length=50, null=True)


class DiamondChild(Parent, DiamondParent):
    pass
`

Reply:
This is expected, it happens because both intermediary parents must define an explicit `OneToOneField(GrandParent, primary_key=True, parent_link=True)` which distinct names.